spec/factories/ext_management_system.rb

```diff
@@ -186,7 +186,7 @@
           :parent => :ems_openstack_infra do
     after :create do |x|
       x.orchestration_stacks << FactoryGirl.create(:orchestration_stack_openstack_infra)
-      4.times { x.hosts << FactoryGirl.create(:host_openstack_infra) }
+      4.times { x.hosts << FactoryGirl.create(:host_openstack_infra, :ext_management_system => FactoryGirl.create(:ems_openstack_infra)) }
```

Comment:
What does this do?  Does the `host_openstack_infra` wind up getting assigned to the `ems_openstack_infra_with_stack` like it did before the change, or does it get assigned to the newly created `ems_openstack_infra`?  If the latter, how is that helpful?

Reply:
What are you trying to do with this factory though?

As it is the host will belong to EMS x, what you're doing here is creating a _new_ EMS

Reply:
In this PR https://github.com/ManageIQ/manageiq-providers-openstack/pull/380/files, I added a callback for updating events after host creation, for some reason, when using this factory, ext_management_system is nil for host.

Reply:
If anything you should set the host's `:ext_management_system => x`

Reply:
ok, so first thing, we should move this factory under OpenStack

now if ext_management_system relation is nil, that would look like rails/factory_firl bug. Lets probably try to remove the code adding hosts entirely and add it to only appropriate spec